### PR TITLE
Add deployment unit tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Similar to #490 

Covered all of the funcs except: 
- `RedefineWith2MiHugepages`
- `RedefineWith1GiHugepages`

### CODE REVIEW

File: tests/utils/deployment/deployment_test.go [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+	deployment.Spec.Template.Spec.Containers = []corev1.Container{}
+
+	RedefineWithContainerSpecs(deployment, []corev1.Container{
+		{
+			Name: "test-container",
+		},
+	})
+	assert.NotNil(t, deployment)
+
+	// Assert that the deployment has a startup probe.
```
Comment (theemj) [inline tests/utils/deployment/deployment_test.go:221]:
```suggestion
	// Assert that the deployment has container specs.
```

File: tests/utils/deployment/deployment_test.go [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+	assert.Equal(t, "150m", deployment.Spec.Template.Spec.Containers[0].Resources.Requests.Memory().String())
+	assert.Equal(t, "100Mi", deployment.Spec.Template.Spec.Containers[0].Resources.Requests.Cpu().String())
+}
+
+func TestRedefineWithRunTimeClass(t *testing.T) {
+	deployment := DefineDeployment("test-deployment", "test-namespace", "test-image", map[string]string{"app": "test"})
+	RedefineWithRunTimeClass(deployment, "test-runtime-class")
+	assert.NotNil(t, deployment)
+
+	// Assert that the deployment has resources set.
```
Comment (theemj) [inline tests/utils/deployment/deployment_test.go:393]:
```suggestion
	// Assert that the deployment has runtime class set.
```

**Review by theemj — APPROVED** [review]
couple of comments about comments, otherwise lgtm!

File: tests/utils/deployment/deployment_test.go (hunk 1 above)
Comment (sebrandon1) [reply tests/utils/deployment/deployment_test.go]: Done!

**Pull request merged**